### PR TITLE
17322 -  shall generate pagination for the frontend even if gateway h…

### PR DESCRIPTION
**Pull request by ko5tik**
…as no database specified

Pagination shall be generated for gateway templates even it gateway itself has no database

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

This PR fixes issue  #17322

### CODE REVIEW

File: generators/entity/index.js [hunk 1]
```diff
@@ -386,7 +386,8 @@ class EntityGenerator extends BaseBlueprintGenerator {
         }
 
         this.entityConfig.name = this.entityConfig.name || context.name;
-        if (![SQL, MONGODB, COUCHBASE, NEO4J].includes(context.databaseType)) {
+        // disable pagination if there is not database, unless it is gateway project
```
Comment (mshima) [inline generators/entity/index.js:389]:
```suggestion
        // disable pagination if there is not database, unless it’s a microservice entity published by a  gateway
```

File: generators/entity/index.js [hunk 2]
```diff
@@ -386,7 +386,8 @@ class EntityGenerator extends BaseBlueprintGenerator {
...
+        if (![SQL, MONGODB, COUCHBASE, NEO4J].includes(context.databaseType) && context.applicationType !== GATEWAY) {
```
Comment (mshima) [inline generators/entity/index.js:390]:
```suggestion
        if (![SQL, MONGODB, COUCHBASE, NEO4J].includes(context.databaseType) && (context.applicationType !== GATEWAY || this.entityConfig.microserviceName)) {
```

**Review by mshima — COMMENTED** [review]
Should be applied to microservice entities only.

File: generators/entity/index.js (hunk 1 above)
Comment (ko5tik) [reply generators/entity/index.js:389]: Agree.  But if there are non-microservice entities  there ought be the database anyway?

Comment (Tcharl) [reply generators/entity/index.js:389]: still, I don't recommend it thin GW with clear responsibility (routing + security) are more maintenable

Comment (ko5tik) [reply generators/entity/index.js:389]: Anyway proposed change is implemented as proposed

Comment (mshima) [reply generators/entity/index.js:389]: It’s missing to update the comment.

Comment (ko5tik) [reply generators/entity/index.js]: pushed

**Review by mshima — APPROVED** [review]
LGTM

**Pull request merged**